src/udev/udevd.c

```diff
@@ -972,6 +963,31 @@ static int event_queue_start(Manager *manager) {
         return 0;
 }
 
+static int event_requeue(Event *event) {
+        usec_t now;
+        int r;
+
+        assert(event);
+        assert(event->manager);
+        assert(event->manager->event);
+
+        event->timeout_warning_event = sd_event_source_disable_unref(event->timeout_warning_event);
+        event->timeout_event = sd_event_source_disable_unref(event->timeout_event);
+
+        event->state = EVENT_QUEUED;
+        if (event->worker && event->worker->event == event)
+                event->worker->event = NULL;
+        event->worker = NULL;
+
+        /* add a short delay to suppress busy loop */
+        r = sd_event_now(event->manager->event, clock_boottime_or_monotonic(), &now);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to get current time, requeueing event without delay: %m");
+
+        event->wait_until = usec_add(now, 200 * USEC_PER_MSEC);
```

Comment:
hmm, so instead of going just by time, wouldn't it make sense to use "secondary" events as trigger for trying again to process the uevent?

by "secondary" events I mean mostly two things: inotify close events and further uevents for the same device?

i.e. let's say a device is locked, then maybe we should just queue the uevent until something later happens, and in that case process the queued uevent first, and then afterwards everything that happened later on it? i.e. use later happenings as reason to reconsider the queued earlier events for the very same device?

In that scheme I think going by timer also makes sense, but more as a safety net to ensure that uevents don#t remain stuck in the queue indefinitely.